File: javascript/gpt/p14.js
// p14: function prompts user for index, returns -1 if out of bounds; program returns 0
const readline = require("readline");

function getValuePrompt(callback) {
  const id_sequence = [10, 20, 30, 40];
  const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
  });
  rl.question("Enter index: ", (answer) => {
    rl.close();
    const idx = parseInt(answer.trim(), 10);
    if (Number.isNaN(idx) || idx < 0 || idx >= id_sequence.length) {
      callback(-1);
    } else {
      callback(id_sequence[idx]);
    }
  });
}

getValuePrompt((val) => {
  console.log(val);
  process.exit(0);
});

File: python/gpt/p14.py
# p14: function prompts user for index, returns -1 if out of bounds; program returns 0


def get_value_prompt():
    id_sequence = [10, 20, 30, 40]
    try:
        idx = int(input("Enter index: "))
    except Exception:
        return -1
    if idx < 0 or idx >= len(id_sequence):
        return -1
    return id_sequence[idx]


def main():
    val = get_value_prompt()
    print(val)
    return 0


if __name__ == "__main__":
    main()
